agent/app/agent.go

```diff
@@ -298,17 +298,36 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 			return exitcodes.ExitTerminal
 		}
 	}
+	numGPUs := int64(0)
 	if agent.cfg.GPUSupportEnabled {
 		err := agent.initializeGPUManager()
 		if err != nil {
 			seelog.Criticalf("Could not initialize Nvidia GPU Manager: %v", err)
 			return exitcodes.ExitError
 		}
+		// Find number of GPUs instance has
+		platformDevices := agent.getPlatformDevices()
+		for _, device := range platformDevices {
+			if *device.Type == ecs.PlatformDeviceTypeGpu {
+				numGPUs++
+			}
+		}
+	}
+
+	hostResources, err := client.GetHostResources()
```

Comment:
Should we move **line 317-321** before **line 301-315** to make it fail fast if agent cannot get host resources?
```
// Get host resources
hostResources, err := client.GetHostResources()
if err != nil {
    seelog.Critical("Unable to fetch host resources")
    return exitcodes.ExitError
}
```
```
// Get GPU devices
numGPUs := int64(0)
	if agent.cfg.GPUSupportEnabled {
		err := agent.initializeGPUManager()
		if err != nil {
			seelog.Criticalf("Could not initialize Nvidia GPU Manager: %v", err)
			return exitcodes.ExitError
		}
		// Find number of GPUs instance has
		platformDevices := agent.getPlatformDevices()
		for _, device := range platformDevices {
			if *device.Type == ecs.PlatformDeviceTypeGpu {
				numGPUs++
			}
		}
	}
```
```
// Update GPU in hostResources map
hostResources["GPU"] = &ecs.Resource{
    Name:         utils.Strptr("GPU"),
    Type:         utils.Strptr("INTEGER"),
    IntegerValue: &numGPUs,
}
```

Reply:
Sounds good to me!